### PR TITLE
[#154442879] Overview of transactions by card

### PR DESCRIPTION
**Pull request by fgiobergia**
PR for [#154442879](https://www.pivotaltracker.com/story/show/154442879), implementing a screen that displays a list of transactions for a given card. 

The credit card preview is a cropped image atm, this will be fixed in [#157422715](https://www.pivotaltracker.com/story/show/157422715)

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Remember to fix the PR title by removing WIP wording when ready
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#154442879](https://www.pivotaltracker.com/story/show/154442879): Come CIT devo poter visualizzare tutti i pagamenti effettuati filtrando per ciascun metodo di pagamento (carta di credito) (UI)

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/screens/wallet/TransactionsScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    );
+  }
+
+  public render(): React.ReactNode {
+    const { params } = this.props.navigation.state;
+    const card: CreditCard = params ? params.card : UNKNOWN_CARD;
+    const transactions: ReadonlyArray<
+      WalletTransaction
+    > = WalletAPI.getTransactions(card.id);
+    const TITLE = I18n.t("wallet.creditDebitCards");
```
Comment (fpersico) [inline ts/screens/wallet/TransactionsScreen.tsx]: As we said put I18n.t directly where needed

**Comment by cloudify** [comment]
@matteodesanti should we use `EUR` or `€` for the currency?

File: ts/components/wallet/CreditCardComponent.tsx [hunk 2]
```diff
@@ -169,7 +170,10 @@ export default class CreditCardComponent extends React.Component<Props> {
               style={CreditCardStyle.rowStyle}
               size={2}
               {...{
-                onPress: (): boolean => navigate("")
+                onPress: (): boolean =>
+                  navigate(ROUTES.WALLET_CARD_TRANSACTIONS, {
```
Comment (cloudify) [inline ts/components/wallet/CreditCardComponent.tsx]: + TODO for later removal?

Comment (fgiobergia) [reply ts/components/wallet/CreditCardComponent.tsx]: Can you elaborate on this comment? This is where the user taps on a CC and navigates to the list of transactions carried out with the selected CC -- any reason why this should be removed later?

Comment (cloudify) [reply ts/components/wallet/CreditCardComponent.tsx]: sorry I thought it was "the shortcut" for reaching the screen, never mind

**Comment by fgiobergia** [comment]
@fpersico does this look master-ready to you?

File: ts/screens/wallet/TransactionsScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+interface OwnProps {
+  readonly navigation: NavigationScreenProp<StateParams>;
+}
+
+type Props = OwnProps & NavigationInjectedProps;
+
+export class TransactionsScreen extends React.Component<Props, never> {
+  constructor(props: Props) {
```
Comment (fpersico) [inline ts/screens/wallet/TransactionsScreen.tsx]: Is this usefull?

Comment (fgiobergia) [reply ts/screens/wallet/TransactionsScreen.tsx]: I would daresay it is not. Getting rid of those

**Pull request merged**